### PR TITLE
Pin kubernetes to <4.0

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Update kubernetes pinning in static requirements files. 

### Tests written?

No - Fixing existing tests

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by s0undt3ch** [comment]
To generate the static requirements:

```
pip install pre-commit
pre-commit -av
```

**Comment by dwoz** [comment]
@s0undt3ch Fixed, thanks!

**Review by s0undt3ch — APPROVED** [review]

**Comment by Ch3LL** [comment]
Do we have an issue open so we can track down why salt does not currently work with 4?

**Comment by s0undt3ch** [comment]
Don't think we have.

**Comment by dwoz** [comment]
re-run centos

**Comment by dwoz** [comment]
re-run full all

**Review by angrydead — COMMENTED** [review]
Can this be simplified to the following?

```
if (name in proc.name() and _search(proc)) or search is None:
    return proc
```

**Comment by dwoz** [comment]
@pullit

It seems like we'd be missing the `if name in proc.name():` after the first if. I think in order for it to be the exact same logic we would need to be something like:

```
if (name in proc.name() and _search(proc)) or (search is None and name in proc.name()):
    return proc
```

However, I also think it's debatable wether or not that's actually simpler. Also, having said that, I'm just trying to prevent some tests from raising and exception in this PR and I feel like changing too much will run the risk of introducing a bug. I'd prefer not to change it.

**Comment by dwoz** [comment]
re-run full all

**Comment by pullit** [comment]
@dwoz That's fair.  I'm new here and don't want to step on any toes.  In hindsight, your original implementation is more readable and scales better if additional functionality is needed.

**Comment by dwoz** [comment]
@pullit Thanks, In any case your review is welcomed. An extra set of eyes no changes never hurts. :)

**Comment by dwoz** [comment]
re-run full all

**Review by s0undt3ch — APPROVED** [review]

**Comment by dwoz** [comment]
re-run full all

**Comment by dwoz** [comment]
re-run full all

File: .ci/kitchen-centos6-py3 [hunk 1]
```diff
@@ -0,0 +1,119 @@
+properties([
```
Comment (s0undt3ch) [inline .ci/kitchen-centos6-py3]: CentOS 6 under Py3?
We're not even building that on branch builds...

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**